### PR TITLE
[fix](ES Catalog)Add array types support in esquery function

### PR DESCRIPTION
**Pull request by qidaye**
## Proposed changes

Support array types in `esquery` function, and add some tests.

### CODE REVIEW

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run external

**Comment by qidaye** [comment]
run feut

**Comment by qidaye** [comment]
run p0

**Comment by qidaye** [comment]
run feut

**Comment by qidaye** [comment]
run p0

**Comment by qidaye** [comment]
run p0

File: gensrc/script/doris_builtins_functions.py [hunk 1]
```diff
@@ -1582,7 +1582,28 @@
         [['esquery'], 'BOOLEAN', ['DATEV2', 'VARCHAR'], ''],
         [['esquery'], 'BOOLEAN', ['DATETIMEV2', 'VARCHAR'], ''],
         [['esquery'], 'BOOLEAN', ['TIMEV2', 'VARCHAR'], ''],
-        [['esquery'], 'BOOLEAN', ['ARRAY', 'VARCHAR'], ''],
+        [['esquery'], 'BOOLEAN', ['ARRAY_BOOLEAN', 'VARCHAR'], ''],
```
Comment (xiaokang) [inline gensrc/script/doris_builtins_functions.py:1585]: use ARRAY<T>

**Comment by qidaye** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Pull request merged**